src/clr-angular/forms/common/providers/layout.service.ts

```diff
@@ -16,11 +16,32 @@ export enum Layouts {
 export class LayoutService {
   layout: Layouts = Layouts.VERTICAL;
 
-  isVertical() {
+  isVertical(): boolean {
     return this.layout === Layouts.VERTICAL;
   }
 
   get layoutClass(): string {
     return `clr-form-${this.layout}`;
   }
+
+  isValid(layout: string): boolean {
+    return this.layoutsAsList().indexOf(layout) > -1;
+  }
+
+  // This is basically a polyfill for Object.values(), which IE11 and Node <9 don't support :(
+  // String enums cannot be reverse-mapped, meaning Layouts['COMPACT'] does not return 'compact' so
+  // this exists to deal with this little caveat to get the list of the values of a string enum.
+  // Source: https://github.com/tc39/proposal-object-values-entries/blob/master/polyfill.js
+  private layoutsAsList(): string[] {
```

Comment:
@gnomeontherun I don't think just to determine `Object.values(Layouts).includes(this.layout)`, we should incorporate external polyfills. This chunk of code looks more like a compiled code. Unless I am overlooking anything, we could do it in simpler and more maintainable ways on our own. One example I could see is here:

```
private layoutValues: string[] = Object.keys(Layouts).map(key => Layouts[key]);

isValid(layout: string): boolean {
    return this.layoutValues.indexOf(layout) > -1;
}
```

Reply:
Thanks, from reading I didn't expect Object.keys to work here either (due to string enum), so this is better.